README.md

```diff
@@ -259,6 +261,12 @@ File extensions are still relevant. While either a `.js` or an `.mjs` file can b
 
 The CommonJS automatic file extension resolution or folder `index.js` discovery are not supported for `import` statements, even when referencing files inside CommonJS packages. Both `import cookies from 'request/lib/cookies'` and `import request from './node_modules/request'` would throw. Automatic file extension resolution or folder `index.js` discovery _are_ still supported for `package.json` `"main"` field specifiers, however, to preserve backward compatibility.
 
+#### Initial entry point
+
+It is outside the scope of this proposal to define how an ESM-supporting Node should determine the parse goal, ESM or CommonJS, for every type of input (file, string via `--eval`, string via `STDIN`, extensionless file, etc.). However, for the purposes of completeness for this proposal we will define _one_ method that we expect Node to support, with the understanding that additional methods will be necessary to handle the other use cases.
+  
+To preserve backward compatibility, we expect that `node file.js` will continue to load the entry point `file.js` as CommonJS by default. (This may be deprecated and eventually changed to an ESM default in the future, but certainly not in the initial release of Node with ESM support.) However, if `file.js` is inside an ESM package scope, Node should load `file.js` as ESM. So for example, if a [`package.json` with an ESM-signifying field](#parsing-packagejson) is in the same folder as `file.js`, `node file.js` would load `file.js` as ESM. The same “search up the file tree for a `package.json`” [algorithm](#procedure) for `import` statements applies when determining the package scope for `file.js`.
```

Comment:
Would it be reasonable to split the last line into a separate paragraph for visibility?

```suggestion
It is outside the scope of this proposal to define how an ESM-supporting Node should determine the parse goal, ESM or CommonJS, for every type of input (file, string via `--eval`, string via `STDIN`, extensionless file, etc.). However, for the purposes of completeness for this proposal we will define _one_ method that we expect Node to support, with the understanding that additional methods will be necessary to handle the other use cases.
  
To preserve backward compatibility, we expect that `node file.js` will continue to load the entry point `file.js` as CommonJS by default. (This may be deprecated and eventually changed to an ESM default in the future, but certainly not in the initial release of Node with ESM support.) However, if `file.js` is inside an ESM package scope, Node should load `file.js` as ESM.

So for example, if a [`package.json` with an ESM-signifying field](#parsing-packagejson) is in the **same** folder as `file.js`, `node file.js` would load `file.js` as ESM. Otherwise it will default to CommonJS.

The proposed way still ensure that `import(…)` statements in either mode for the main entry point still applies the usual “search up the file tree for a `package.json`” [algorithm](#procedure) for `import` statements. A more complete solution should be considered that would mitigate `package.json` complexities for new user while allowing them to opt into either modes when running `node file.js`.
```

Reply:
So you’re proposing adding this?

> The proposed way still ensure that `import(…)` statements in either mode for the main entry point still applies the usual “search up the file tree for a `package.json`” [algorithm](#procedure) for `import` statements. A more complete solution should be considered that would mitigate `package.json` complexities for new user while allowing them to opt into either modes when running `node file.js`.

You can’t use `import()` for the main entry point. The main entry point is just whatever file is passed to `node` to run. Inside that file we can have `import` or `import()`, but this section is discussing what happens before that point.

I also don’t think there’s some better more complete solution out there. This _is_ the best solution for traditional apps, that have a `package.json` with a `dependencies` and so on. The only use cases that aren’t satisfied well by this are shell-like scripts that have no dependencies, or uses like `--eval` or `STDIN`. Those will need to be handled separately regardless.

Reply:
I'm trying to say that file.js can safely technically avoid esm/cjs scope complexicities if for the entry point it was basically possible to `import('./esm.js')` or `require('./cjs.js')`;

So entry point not having `package.json` **directly** next to it would default to 'cjs' but if it `import('./esm.js')`:

1. in it's parent folder is an ESM scope then it READ_BOUNDARY and it's okay
2. in it's parent folder is an CJS scope then it READ_BOUNDARY and throws (catchable rejection).

*Side Note*: I am thinking of a good way for the entry to easily recover if the environment is influenced by a parent `package.json` as Myles points out and that could be  `import('./esm.js').catch(() => require('./cjs.js'))`. So my thought is main versus the effective entry point as two distinct thing here.

Reply:
@SMotaal I think you’re talking about the code inside that initial entry point file, right? Doesn’t `import()` always import as ESM? It’s only `import` (no parentheses) that I thought we were thinking of building CommonJS interop into.